### PR TITLE
email not case-sensitive, better validation, user-control page

### PR DESCRIPTION
**Pull request by thedekel**
see notes on the latest commit, and title

### CODE REVIEW

**Comment by thelastnode** [comment]
Pull requests really need to be smaller than this. I see 3 - 4 features in this request - this makes it tough to review and approve individual features.

On a more positive note, this is good stuff, glad this is being actively maintained

**Comment by thedekel** [comment]
Don't mind the Travis thing failing, I've just been playing around with setting up build validation in preparation for when we add all those tests and stuff, I'll re-enable it once it's properly set-up.

File: lib/auth.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 var bcrypt    = require('bcrypt');
 var everyauth = require('everyauth');
-var mongoose  = require('mongoose');
-var models = require('./models');
+var mongoose  = require('mongoose'); var models = require('./models');
```
Comment (Magicjarvis) [inline lib/auth.js]: %s/; /;\r/g

File: lib/auth.js [hunk 2]
```diff
@@ -63,50 +62,59 @@ exports.validateRegistration = function validateRegistration(attribs) {
 };
 
 exports.registerUser = function registerUser(attribs) {
-    var promise = this.Promise();
-    
-    bcrypt.genSalt(function(err, salt) {
+  var promise = this.Promise(), emailre, gtidre, standardre, passwordre;
+  //we really should have had some validation right here...
+  emailre = /[a-zA-Z0-9-_]+@[a-zA-Z0-9-_]+\.[a-zA-Z0-9-_]+/;
```
Comment (Magicjarvis) [inline lib/auth.js]: perhaps pull these things out into a util file or something.

File: public/javascripts/userman.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  ulistdiv.appendChild(table);
+  for (i = 0; i < users.length; i += 1){
+    usertr = createTr(users[i]);
+    table.appendChild(usertr);
+  }
+  document.getElementsByClassName("content")[0].appendChild(table);
+  regfield = document.getElementById('regfield');
+  regfield.oninput= function(e){
+    var re = new RegExp(regfield.value, 'i'), i, tr;
+    for (i = 4; i < table.children.length; i += 1){
```
Comment (Magicjarvis) [inline public/javascripts/userman.js]: magic numbers

File: routes/admin.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
-var async = require('async');
-var models = require('../lib/models');
+var async = require('async'),
```
Comment (Magicjarvis) [inline routes/admin.js:1]: we really should stick to one standard. 

Either:
var a = ...;
var b = ...; 

Or:
var a = ...,
      b = ...;

Not both.

File: routes/events.js [hunk 5]
```diff
@@ -445,11 +445,12 @@ exports.guest = function(req, res, next) {
  */
 exports.guest_attend = function(req, res, next) {
   //validation regex for email and gtids to filter out bad data
-  var emailre = /\S+@\S+\.\S+/;
-  var gtidre = /\d{9}/;
+  var emailre = /[a-zA-Z0-9-_]+@[a-zA-Z0-9-_]+\.[a-zA-Z0-9-_]+/;
```
Comment (Magicjarvis) [inline routes/events.js]: util file is looking really good right now.

**Comment by Magicjarvis** [comment]
Make those minor changes and I'll merge.

**Pull request merged**